Src/DependencyCollector/Shared/Implementation/HttpCoreDiagnosticSourceListener.cs

```diff
@@ -36,11 +37,11 @@ internal class HttpCoreDiagnosticSourceListener : IObserver<KeyValuePair<string,
         private readonly PropertyFetcher deprecatedResponseFetcher = new PropertyFetcher("Response");
         private readonly PropertyFetcher deprecatedRequestGuidFetcher = new PropertyFetcher("LoggingRequestId");
         private readonly PropertyFetcher deprecatedResponseGuidFetcher = new PropertyFetcher("LoggingRequestId");
-        
+
         #endregion
 
-        private readonly ConcurrentDictionary<string, IOperationHolder<DependencyTelemetry>> pendingTelemetry =
-            new ConcurrentDictionary<string, IOperationHolder<DependencyTelemetry>>();
+        private readonly ConditionalWeakTable<HttpRequestMessage, IOperationHolder<DependencyTelemetry>> pendingTelemetry = 
```

Comment:
You are right, we can call begin and then null request and after Gc.Collect we should expect the table to be empty. But my point: what are we trying to test here? ConditionalWeakTable? The fact that HttpRequestMessage is collected at some point? Those are not part of AppInsights.